Build and deploy nvcc versions

jakirkham opened the pull request:
Builds out all of the version of the `nvcc` compiler package. Supersedes previous attempts by building all CUDA versions in one branch.

Edit: FYI build number does not need to change as this has yet to be published.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

Closes https://github.com/conda-forge/nvcc-feedstock/pull/5
Closes https://github.com/conda-forge/nvcc-feedstock/pull/4

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

kkraus14 commented on recipe/conda_build_config.yaml:
```diff
@@ -0,0 +1,11 @@
+  - 10.0
+  - 9.2
+zip_keys:
+  - - CUDA_VER
```
Should there be two dashes here?

jakirkham replied:
Yeah the second dash means this is a group of things that should be zipped together, but it should not be zipped with anything else being zipped. For example we [zip `python` (and `vc` on Windows)]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/1730aa4ed6f04bea8479e17e37de6598d92bb936/recipe/conda_build_config.yaml#L135-L139 ), which should be treated as a separate group.

Further conda-build requires that if we use nested lists, we must use them in all places. Otherwise it is an error.

ref: https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#coupling-keys

jakirkham commented:
@isuruf, could you please give this one a look? It replaces the other two PRs and consolidates the versions as discussed.

jakirkham commented:
Going ahead and merging to move things forward here. AIUI this addressed the concerns raised in the other PRs. That said, if other issues crop up, am happy to iterate on these with people. Thanks all for the reviews and help here. 😀
